Sofa/Component/Mapping/Linear/tests/DistanceToPlaneMapping_test.cpp

```diff
@@ -0,0 +1,311 @@
+using sofa::component::statecontainer::MechanicalObject ;
+
+#include <sofa/simulation/Node.h>
+#include <sofa/linearalgebra/EigenSparseMatrix.h>
+#include <sofa/core/trait/DataTypes.h>
+
+#define EPSILON 1e-12
+
+using sofa::defaulttype::Vec1Types;
+
+template <class DataType>
+class PlaneMappingTest : public testing::Test
+{
+
+    typedef sofa::type::Vec<sofa::Deriv_t<DataType>::spatial_dimensions,typename sofa::Deriv_t<DataType>::value_type> PlaneNormalType;
+    typedef sofa::type::Vec<sofa::Coord_t<DataType>::spatial_dimensions,typename sofa::Coord_t<DataType>::value_type> PlanePointType;
+public:
+    sofa::simulation::Simulation* createSimpleScene(typename sofa::component::mapping::linear::DistanceToPlaneMapping<DataType>::SPtr mapping)
+    {
+
+
+        sofa::simulation::Simulation* simu = sofa::simulation::getSimulation();
+
+        const Node::SPtr node = simu->createNewGraph("root");
+
+        typename MechanicalObject<DataType>::SPtr mechanical = New<MechanicalObject<DataType>>();
+        mechanical->resize(10);
+        auto inPos = mechanical->writePositions();
+        inPos.resize(10);
+        auto inRestPos = mechanical->writeRestPositions();
+        inRestPos.resize(10);
+        node->addObject(mechanical);
+
+
+        const Node::SPtr nodeMapping = node->createChild("nodeToMap");
+        typename MechanicalObject<sofa::defaulttype::Vec1Types>::SPtr targetMechanical = New<MechanicalObject<sofa::defaulttype::Vec1Types>>();
+        nodeMapping->addObject(targetMechanical);
+        mapping->setFrom(mechanical.get());
+        mapping->setTo(targetMechanical.get());
+        nodeMapping->addObject(mapping);
+
+        EXPECT_NO_THROW(
+            sofa::simulation::node::initRoot(node.get())
+        );
+        return simu;
+    }
+
+    PlaneNormalType getPseudoRandomNormal()
+    {
+        constexpr std::array<double,6> randomValuesForNormal = {1, 5.1, -6, 0, 9.5, 4};
+        PlaneNormalType returnVec;
+        for (unsigned i = 0; i< sofa::Deriv_t<DataType>::spatial_dimensions; ++i)
+        {
+            returnVec[i] = randomValuesForNormal[i];
+        }
+        return returnVec;
+    }
+
+
+    PlanePointType getPseudoRandomPoint()
+    {
+        constexpr std::array<double,7> randomValuesForPoint = {-2.5, 1.4, 3, 0, 0.7, 12, 7.07};
+        PlanePointType returnVec;
+        for (unsigned i = 0; i< sofa::Coord_t<DataType>::spatial_dimensions; ++i)
+        {
+            returnVec[i] = randomValuesForPoint[i];
+        }
+        return returnVec;
+
+    }
+
+    void testInit(PlaneNormalType planeNormal)
+    {
+        typename sofa::component::mapping::linear::DistanceToPlaneMapping<DataType>::SPtr mapping = New<sofa::component::mapping::linear::DistanceToPlaneMapping<DataType>>();
+        mapping->d_planeNormal.setValue(planeNormal);
+        auto simu = this->createSimpleScene(mapping);
+        EXPECT_LE(fabs(mapping->d_planeNormal.getValue().norm() - 1.0),EPSILON);
```

Comment:
I don't know what is the precision of this kind of computation here. My impression is that fabs() return the type of norm() (so it can be either double or float)... which is then compared to EPSILON... which is a double.